### PR TITLE
Feature/vault ssl

### PR DESCRIPTION
**Pull request by ChrisAubuchon**

### CODE REVIEW

File: roles/vault/templates/vault.hcl.j2 [hunk 1]
```diff
@@ -3,11 +3,11 @@ backend "consul" {
 	path = "vault"
 	scheme = "http"
 	token = "{{ consul_acl_master_token }}"
-	advertise_addr = "{{ private_ipv4 }}"
+	advertise_addr = "https://{{ inventory_hostname }}.node.{{ consul_dns_domain }}:{{ vault_default_port }}"
 }
 
 listener "tcp" {
-	address = "127.0.0.1:8200"
+	address = "0.0.0.0:8200"
```
Comment (ryane) [inline roles/vault/templates/vault.hcl.j2]: should we use `vault_default_port` here too?

File: terraform.sample.yml [hunk 2]
```diff
@@ -25,10 +25,6 @@
     # this variable.
     # consul_acl_datacenter: your_primary_datacenter
 
-    # consul_dns_domain is repeated across these plays so that all the
-    # components know what information to use for this values to help with
-    # service discovery.
-    consul_dns_domain: consul
```
Comment (ryane) [inline terraform.sample.yml]: we should remove the re-definition of `consul_dns_domain` from `addons/glusterfs.yml` and `kubernetes.yml` too

File: roles/vault/templates/vault.hcl.j2 (hunk 1 above)
Comment (ChrisAubuchon) [inline roles/vault/templates/vault.hcl.j2]: If we're going to use `vault_default_port` then we should use it here too.

**Comment by ryane** [comment]
this is almost there but there is one remaining problem. The vault distributive tests are failing with:

```
time="2016-02-01T14:29:46Z" level=warning msg="Check(s) failed, printing checklist report" checklist="vault health check" report="↴\nTotal: 4\nPassed: 2\nFailed: 2\nNo such file or directory: /usr/lib/systemd/system/vault.service\nPort not open:\n\tSpecified: 8200\n\tActual: 15050, 47472, 8400, 51601, 8500, 8500, 53, 22, 8600, 25, 8500, 36038, 51355, 8500, 8500, 8500, 8500, 8500, 8500, 37679, 8500, 8500, 8500, 15050, 37666, 15050, 8500, 8500, 8500, 8500, 49076, 34935, 46235, 49496, 37670, 8500, 37351, 8500, 8500, 8500, 8500, 8500, 8500, 8500, 8500, 37684, 46971, 34817, 8500, 34835"
```

port 8200 is definitely listening. I suspect the problem may be due to https://github.com/CiscoCloud/distributive/issues/127

**Comment by ryane** [comment]
tested on aws. consul is all green and vault is available:

```shell
$ export VAULT_ADDR=https://consul.service.consul:8200
$ vault auth
Token (will be hidden):
Successfully authenticated!
token: <token>
token_duration: 0
token_policies: [root]
$ vault write secret/test value=test
Success! Data written to: secret/test
$ vault read secret/test
Key             Value
lease_duration  2592000
value           test
```

**Pull request merged**